spring-cloud-gcp-data-spanner/src/test/java/org/springframework/cloud/gcp/data/spanner/repository/it/ParallelSpannerRepositoryIntegrationTests.java

```diff
@@ -0,0 +1,80 @@
+/*
+ * Copyright 2017-2019 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.cloud.gcp.data.spanner.repository.it;
+
+import java.util.List;
+import java.util.function.IntConsumer;
+import java.util.stream.IntStream;
+
+import com.google.cloud.spanner.KeySet;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.cloud.gcp.data.spanner.test.AbstractSpannerIntegrationTest;
+import org.springframework.cloud.gcp.data.spanner.test.domain.Trade;
+import org.springframework.cloud.gcp.data.spanner.test.domain.TradeRepository;
+import org.springframework.test.context.junit4.SpringRunner;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+/**
+ * Tests multiple threads using a single repository instance.
+ *
+ * @author Chengyuan Zhao
+ */
+@RunWith(SpringRunner.class)
+public class ParallelSpannerRepositoryIntegrationTests extends AbstractSpannerIntegrationTest {
+
+	private final static int PARALLEL_OPERATIONS = 20;
```

Comment:
Why 20? Looks like a magic number.

Reply:
It's an arbitrary number, and I don't think one could make a real argument for any number here. I'm open to any number that's not too small :)